spec/var-ref.test.sh

```diff
@@ -23,6 +29,154 @@ myfunc '?'  # osh doesn't do this dynamically
 ## stdout-json: "myfunc\n0\n"
 ## N-I mksh stdout-json: "ref\nref\n"
 
+#### indirection, *then* fancy expansion features
+check_() {
+    [ "$1" = "$2" ] || { echo "$1 vs $2"; }
+}
+check() {
```

Comment:
Yeah, reasonable. :-) Will change.

One thought I was playing with as I wrote these was that something like these functions (with the nice printing of input plus actual vs. expected when things fail) could be useful in a broader range of tests, for things like expansions where the content of a single test case can be super short like the cases below are. Possibly as shell functions that are just pre-supplied by the test harness (opt-in per test file?), or possibly as some logic more directly in the harness (invoked by a variation of the `####` syntax?) -- not sure what would work out better.

One advantage of doing something like that is that it would make it possible to do lots of exhaustive cases in a style like this, without requiring a few lines of this code after each `####` to set it up -- which would help a lot in splitting them up! :smile: 

Anyway, so one thing these `check` / `err` are doing is experimenting with test notation, with that in mind.

Reply:
I went for `check_eq` / `check_expand`. I'm inclined to try to keep the names pretty short -- I think ergonomics like that are helpful for getting to a good comprehensive set of cases.